eitprocessing/features/breath_detection.py

```diff
@@ -0,0 +1,471 @@
+        window_size = int(self.sample_frequency * self.averaging_window_duration)
+        averager = MovingAverage(window_size=window_size, window_function=self.averaging_window_function)
+        moving_average = averager.apply(data)
+
+        peak_indices = self._find_features(data, moving_average)
+        valley_indices = self._find_features(data, moving_average, invert=True)
+
+        peak_indices, valley_indices = self._remove_edge_cases(data, peak_indices, valley_indices, moving_average)
+        peak_indices, valley_indices = self._remove_doubles(data, peak_indices, valley_indices)
+        peak_indices, valley_indices = self._remove_low_amplitudes(data, peak_indices, valley_indices)
+        return peak_indices, valley_indices
+
+    def _find_features(
+        self,
+        data: np.ndarray,
+        moving_average: np.ndarray,
+        invert: float = False,
+    ) -> np.ndarray:
+        """
+        Find features (peaks or valleys) in the data.
+
+        This method finds features (either peaks or valleys) in the data using
+        the `scipy.signal.find_peaks()` function. The minimum distance (in
+        time) between peaks is determined by the `minimum_duration` attribute.
+
+        To find peaks, `invert` should be False. To find valleys, `invert`
+        should be True, which inverts the data before finding peaks.
+
+        Args:
+            data (np.ndarray): a 1D array containing the data.
+            moving_average (NDArrag): a 1D array containing the moving average
+                of the data.
+            invert (float, optional): whether to invert the data before
+            detecting peaks. Defaults to False.
+
+        Returns:
+            np.ndarray: a 1D-array containing the indices of the features.
+        """
+        data_ = -data if invert else data
+        moving_average_ = -moving_average if invert else moving_average
+        feature_indices, _ = signal.find_peaks(
+            data_,
+            distance=self.minimum_duration * self.sample_frequency,
+            height=moving_average_,
+        )
+
+        return feature_indices
+
+    def _remove_edge_cases(
+        self,
+        data: np.ndarray,
+        peak_indices: np.ndarray,
+        valley_indices: np.ndarray,
+        moving_average: np.ndarray,
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """
+        Remove overdetected peaks/valleys at the start and end of the data.
+
+        This method removes a valley at the start of the data if the data before said valley stays below the moving
+        average of the data at said valley. Likewise, it removes the last valley if the data after the last valley stays
+        below the moving average of the data at said valley. This ensures a valley is a true valley, and not just a
+        local minimum while the true valley is cut off.
+
+        Then, all peaks that occur before the first and after the last valley
+        are removed. This ensures peaks only fall between valleys.
```

Comment:
There seem to be a few different standards for the width/indentation of the docstrings, and these are not the same as the width. I don't think this is a huge deal, but would be nice to fix up cleanly. I'm not sure there's a ruff formatting rule that would automatically fix this (but there might be).
(the next comment (re line 202 and 204) is another example.

@psomhorst , i believe you have some standardized setting for this in VS code. If you copy those settings to the .vscode file, then it would be easier for everyone to adhere to a single style, even without the formatter fixing it for us.

Reply:
I ma de the line length consistent for the files relevant to this PR.